compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -443,5 +445,12 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
      */
     private def normalizeErasedRhs(rhs: Tree, sym: Symbol)(using Context) =
       if (sym.isEffectivelyErased) dropInlines.transform(rhs) else rhs
+
+    private def annotateExperimental(sym: Symbol)(using Context): Unit =
+      if sym.is(Enum) && sym.hasAnnotation(defn.ExperimentalAnnot) then
+        // Add @experimental annotation to enum class definitions
```

Comment:
Why is this needed if we also have `checkExperimentalInheritance`?

Reply:
This is not needed anymore. Removed the code. Though now there is similar looking logic for `case class` modules.